src/OpenTelemetry.Extensions.PersistentStorage.Abstractions/Storage.cs

```diff
@@ -0,0 +1,71 @@
+﻿// <copyright file="Storage.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Extensions.PersistentStorage.Abstractions
+{
+    /// <summary>
+    /// Represents persistent storage.
+    /// </summary>
+    public abstract class Storage
```

Comment:
Name feels a little generic. Some suggestions: `PersistentBlobStorage`, `PersistentBlobProvider`, `BlobProvider`

I threw "provider" in there because that is what [IFileProvider](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.fileproviders.ifileprovider?view=dotnet-plat-ext-6.0) does 🤷

Reply:
I like PersistentBlobProvider for this. The other one as PersistentBlob.
@reyang - Do you have any preferences?